source/agora/node/Registry.d

```diff
@@ -413,7 +413,7 @@ public class NameRegistry: NameRegistryAPI
                         goto BAILOUT;
                     }
                     // Ignore the DO bit for now
-                    payloadSize = min(opt.payloadSize(), ushort(512));
+                    payloadSize = max(opt.payloadSize(), ushort(512));
```

Comment:
I remember looking at this `min` long and hard and having a very good reason for it. Should have added a comment.

Looking at the code, it was to avoid DOS: If we take the client's value at face value, we essentially allow any size of response to be written. Using `min` ensures that we never send messages higher than 512 bytes.

So instead, this should be: `min(max(opt.payloadSize(), ushort(512)), OPTRR.init.payloadSize())`.